salt/_states/metalk8s_package_manager.py

```diff
@@ -0,0 +1,55 @@
+"""Custom state for managing yum packages and dependencies"""
+
+from __future__ import absolute_import
+
+import logging
+
+from salt.exceptions import CommandExecutionError
+
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = "metalk8s_package_manager"
+
+
+def __virtualname__():
+    try:
+        os_family = __grains__['os_family'].lower()
+    except Exception:
+        return (False, "No yum based system detected")
+
+    if 'metalk8s_package_manager.list_pkg_deps' not in __salt__:
+        return (False, '`metalk8s_package_manager.list_pkgs_deps` not found')
+
+    if os_family == 'redhat':
+        return __virtualname__
+    return(False, "Metalk8s_package_manager: no yum based system detected")
+
+
+def pkg_deps(name, version):
```

Comment:
This is definitly not a state. 
The state should get the pkg_deps using the module and use it directly in a classic `pkg.installed`